### PR TITLE
Add Frequency re-sampling

### PR DESCRIPTION
**Pull request by vinayphadnis**
Add code to change the frequency to the desired (16 kHz) from any pre-specified microphone sample rate.
Solves #1279 

**Approach:**
Calculating the LCM (Least Common Multiple) of the Microphone sample rate and the expected sample rate(16 kHz)
**Upscaling** the Microphone chunk array by duplicating adjacent values.
**Downscaling** the upscaled array to the desired frequency by calculating mean to approximate closely without loss of generality

**Example**
Consider a very common microphone frequency 44.1 kHz or 44100
The LCM of 44100 and 16000 is 7056000

In simple terms:
44100 * 160 = 7056000: Every element in the input array is repeated 160 times in the upscaled array,
16000 * 441 = 7056000: The mean of every 441 elements comprise of a single element in the downscaled array
This ensures that regardless of the input sampling rate, the re-sampled array always has a length of 16000

### CODE REVIEW

**Comment by okuchaiev** [comment]
Thank you very much for this PR! Could you please sign your committed commits. DCO bot's output tells you how to do that https://github.com/NVIDIA/NeMo/pull/1281/checks?check_run_id=1244379368

**Comment by vsl9** [comment]
Thanks for the PR! It is a nice example of very fast signal resampling.

But I wouldn't recommend to use it due to potential degradation in audio quality and transcription accuracy:
Upsampling by duplicating samples and downsampling by averaging samples will result in severe aliasing in spectrum. 
It might be ok for neural model if it was trained with such distortions. But we employed high-quality sinc-based resampling for all signal preprocessing before training. So while ASR model might produce correct transcriptions on audio with aliasing, it is definitely not optimal.

**Comment by blisc** [comment]
Thanks, for your changes. However as per @vsl9's advice, I will close this PR.
